core/src/main/java/com/xatkit/core/recognition/nlpjs/NlpjsIntentRecognitionProvider.java

```diff
@@ -0,0 +1,210 @@
+                .intents(new ArrayList<>(this.intentsToRegister.values()))
+                .entities(new ArrayList<>(this.entitiesToRegister.values()))
+                .build();
+        try {
+            boolean isDone = false;
+            int attemptsLeft = 10;
+            this.nlpjsService.trainAgent(agentId, trainingData);
+            while (!isDone && attemptsLeft > 0) {
+                Thread.sleep(2000);
+                Agent agent = this.nlpjsService.getAgentInfo(this.agentId);
+                if (agent.getStatus().equals(AgentStatus.READY))
+                    isDone = true;
+                else
+                    attemptsLeft--;
+            }
+            if (isDone)
+                Log.info("NLP.js agent trained.");
+            else
+                throw new IntentRecognitionProviderException("Failed to train the NLP.js agent");
+        } catch (IOException | InterruptedException e) {
+            throw new IntentRecognitionProviderException("An error occurred during the NLP agent training: ", e);
+        }
+    }
+
+    @Override
+    public XatkitSession createContext(@NonNull String sessionId) throws IntentRecognitionProviderException {
+        return new XatkitSession(sessionId);
+    }
+
+    @Override
+    public void shutdown() throws IntentRecognitionProviderException {
+
+    }
+
+    @Override
+    public boolean isShutdown() {
+        return nlpjsService.isShutdown();
+    }
+
+    @Override
+    protected RecognizedIntent getIntentInternal(@NonNull String input, @NonNull StateContext context) throws IntentRecognitionProviderException {
+        checkNotShutdown();
+        checkArgument(!input.isEmpty(), "Cannot retrieve the intent from empty string");
+        try {
+            UserMessage userMessage = new UserMessage(input);
+            RecognitionResult recognitionResult = this.nlpjsService.getIntent(agentId, userMessage);
+            List<Classification> classifications = recognitionResult.getClassifications();
+            if (!classifications.isEmpty() && classifications.get(0).getIntent().equals("None")) {
+                RecognizedIntent recognizedIntent = IntentFactory.eINSTANCE.createRecognizedIntent();
+                recognizedIntent.setDefinition(DEFAULT_FALLBACK_INTENT);
+                recognizedIntent.setRecognitionConfidence(recognitionResult.getScore());
+                recognizedIntent.setMatchedInput(recognitionResult.getUtterance());
+                return recognizedIntent;
+            }
+            List<RecognizedIntent> recognizedIntents = nlpjsRecognitionResultMapper.mapRecognitionResult(recognitionResult);
+            RecognizedIntent recognizedIntent = getBestCandidate(recognizedIntents, context);
+            recognizedIntent.getOutContextInstances().addAll(nlpjsRecognitionResultMapper.mapParamterValues(recognizedIntent, recognitionResult.getEntities()));
+            return recognizedIntent;
```

Comment:
I miss a call to RecognitionMonitor#logRecognizedIntent here.

Reply:
I'll check that